vspace/src/main/java/edu/asu/diging/vspace/web/staff/ListFileController.java

```diff
@@ -0,0 +1,37 @@
+
+package edu.asu.diging.vspace.web.staff;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.vspace.core.model.impl.VSFile;
+import edu.asu.diging.vspace.core.services.impl.FileManager;
+
+@Controller
+public class ListFileController {
+        
+    @Autowired
+    private FileManager fileManager;
+    
+    @RequestMapping(value = "/staff/files/list", method = RequestMethod.GET)
+    public String getFilesList(Model model, @RequestParam(value = "filesPagenum", required = false, defaultValue = "1") String filesPagenum) {
+               
+        Page<VSFile> filesPage = fileManager.getAllFiles(Integer.parseInt(filesPagenum));
```

Comment:
if `filesPagenum` isn't a number this will throw an exception

Reply:
Since we are setting a default value to "1", will there be a case when it won't be number?

Reply:
since it's a request parameter you can't control what's being send to the server. Could be anything. default value is only used when no parameter is sent.

Reply:
okay yes. got it.